sdk/src/types/block/semantic/mod.rs

```diff
@@ -13,32 +13,34 @@ use primitive_types::U256;
 pub use self::{error::TransactionFailureReason, state_transition::StateTransitionVerifier};
 use crate::types::block::{
     address::Address,
-    context_input::{BlockIssuanceCreditContextInput, CommitmentContextInput, RewardContextInput},
-    output::{AccountId, AnchorOutput, ChainId, FoundryId, Output, OutputId, TokenId},
-    payload::signed_transaction::{Transaction, TransactionCapabilityFlag, TransactionSigningHash},
+    context_input::RewardContextInput,
+    output::{feature::Features, AccountId, AnchorOutput, ChainId, FoundryId, Output, OutputId, TokenId},
+    payload::signed_transaction::{Transaction, TransactionCapabilityFlag, TransactionId, TransactionSigningHash},
     protocol::ProtocolParameters,
+    slot::SlotCommitmentId,
     unlock::Unlock,
     Error,
 };
 
 ///
 pub struct SemanticValidationContext<'a> {
     pub(crate) transaction: &'a Transaction,
+    pub(crate) transaction_id: TransactionId,
     pub(crate) transaction_signing_hash: TransactionSigningHash,
     pub(crate) inputs: &'a [(&'a OutputId, &'a Output)],
     pub(crate) unlocks: Option<&'a [Unlock]>,
     pub(crate) input_amount: u64,
     pub(crate) input_mana: u64,
     pub(crate) mana_rewards: BTreeMap<OutputId, u64>,
+    pub(crate) commitment_context_input: Option<SlotCommitmentId>,
     pub(crate) reward_context_inputs: HashMap<OutputId, RewardContextInput>,
-    pub(crate) commitment_context_input: Option<CommitmentContextInput>,
-    pub(crate) bic_context_input: Option<BlockIssuanceCreditContextInput>,
     pub(crate) input_native_tokens: BTreeMap<TokenId, U256>,
     pub(crate) input_chains: HashMap<ChainId, (&'a OutputId, &'a Output)>,
     pub(crate) output_amount: u64,
     pub(crate) output_mana: u64,
     pub(crate) output_native_tokens: BTreeMap<TokenId, U256>,
     pub(crate) output_chains: HashMap<ChainId, (OutputId, &'a Output)>,
+    pub(crate) block_issuer_mana: HashMap<AccountId, (u64, u64)>,
```

Comment:
I think code readability could profit from replacing this u64 tuple to a struct with some named fields, but I would understand if you think it's not worth the effort.